File: Tutorial5-1Dagger-Android-ContributesAndroidInjector/src/main/java/com/example/tutorial5_1dagger_android_contributesandroidinjector/DummyDependency.java
package com.example.tutorial5_1dagger_android_contributesandroidinjector;

import android.content.Context;

public class DummyDependency {

    private Context context;

    public DummyDependency(Context context) {
        this.context = context;
        System.out.println("DummyDependency Context: " + context);
    }

    public String getApplicationName() {
        return context.getString(R.string.app_name);
    }

}

File: Tutorial5-1Dagger-Android-ContributesAndroidInjector/src/main/java/com/example/tutorial5_1dagger_android_contributesandroidinjector/DummyDependency.kt
package com.example.tutorial5_1dagger_android_contributesandroidinjector

import android.content.Context

class DummyDependency(private val context: Context) {

    val applicationName: String
        get() = context.getString(R.string.app_name)

    init {
        println("DummyDependency Context: $context")
    }
}